src/OpenIddict.EF/OpenIddictContext.cs

```diff
@@ -10,12 +10,13 @@
 using OpenIddict.Models;
 
 namespace OpenIddict {
-    public class OpenIddictContext<TUser, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
+    public class OpenIddictContext<TUser, TApplication, TRole, TKey> : IdentityDbContext<TUser, TRole, TKey>
         where TUser : IdentityUser<TKey>
+        where TApplication : Application
         where TRole : IdentityRole<TKey>
         where TKey : IEquatable<TKey> {
-        public DbSet<Application> Applications { get; set; }
+        public DbSet<TApplication> Applications { get; set; }
     }
 
-    public class OpenIddictContext<TUser> : OpenIddictContext<TUser, IdentityRole, string> where TUser : IdentityUser { }
+    public class OpenIddictContext<TUser> : OpenIddictContext<TUser, Application, IdentityRole, string> where TUser : IdentityUser { }
```

Comment:
mmmm ok, I have some applications that use IdentityUser as is, but I see your point (also passing IdentityUser is trivial in case of not using a custom type. let's keep it this way and if we see someone ask for it we can always add it later.